src/vdom/component.js

```diff
@@ -61,38 +53,35 @@ export function setComponentProps(component, props, renderMode, context, mountAl
  * Render a Component, triggering necessary lifecycle events and taking
  * High-Order Components into account.
  * @param {import('../component').Component} component The component to render
+ * @param {object} props The next props
+ * @param {object} context The next context
  * @param {number} [renderMode] render mode, see constants.js for available options.
  * @param {boolean} [mountAll] Whether or not to immediately mount all components
  * @param {boolean} [isChild] ?
  * @private
  */
-export function renderComponent(component, renderMode, mountAll, isChild) {
+export function renderComponent(component, props, context, renderMode, mountAll, isChild) {
 	if (component._disable) return;
 
-	let props = component.props,
-		state = component.state,
-		context = component.context,
-		previousProps = component.prevProps || props,
-		previousState = component.prevState || state,
-		previousContext = component.prevContext || context,
+	let state = component._nextState,
+		previousState = component.state,
+		previousProps = component.props,
+		snapshot = component.context,
 		isUpdate = component.base,
 		nextBase = component.nextBase,
 		initialBase = isUpdate || nextBase,
 		initialChildComponent = component._component,
 		skip = false,
-		snapshot = previousContext,
-		rendered, inst, cbase;
+		rendered, inst, cbase,
+		getDerived = component.constructor.getDerivedStateFromProps;
```

Comment:
this will invoke `getDerivedStateFromProps` with a `this` value of `undefined`.

Reply:
That was one of my byte-shavings, sad but true.

Reply:
The really sad part is that is current behavior in 8.3.0
https://github.com/developit/preact/blob/e833786fb9543fd7becd561e59ae9132e7533dbd/src/vdom/component.js#L87
That line is a refactor that does not change behavior. This is a bug from the master branch, not this PR.

Reply:
Funny, I thought getDerivedStateFromProps was supposed to be a static method, and I don't know what the value of `this` should be if there is no instance.

https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
I don't think it's a bug after all, in my PR or master!

Reply:
The current `master` will invoke getDerivedStateFromProps() with `this` pointing to the constructor. I am not sure if that's something people use, but it's what React does.